app/auth/security.py

```diff
@@ -0,0 +1,93 @@
+from typing import Union, Annotated
+from datetime import timezone, timedelta, datetime
+
+from passlib.context import CryptContext
+import jwt
+from jwt.exceptions import InvalidTokenError
+from fastapi.security import OAuth2PasswordBearer
+from fastapi import HTTPException, Depends, status
+from ..models.users import Users, TokenData
+from app.db import engine
+from sqlmodel import Session
+from app.config import Config
+
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+def verify_password(plain_password, hashed_password):
+    return pwd_context.verify(plain_password, hashed_password)
+
+def get_password_hash(password):
+    return pwd_context.hash(password)
+
+def get_user(username: str):
+    with Session(engine) as session:
+        
+        user = session.get(Users, username)
+        
+        return user
+
+def authenticate_user(username: str, password: str):
+    """
+    This function returns the user if they are authenticated against their
+    hashed password and exist in the database. Used in service login.
+
+    Args:
+        username: Username data as a string
+        password: Password data as a string
+
+    Returns:
+        user: A string that contains the user information to be used
+        to create the access token
+
+    Raises:
+        False: If user does not exist or if the verify password function
+        cannot match the current password with the hashed user password
+    """
+    user = get_user(username)
+    if not user:
+        return False
+    if not verify_password(password, user.hashed_password):
+        return False
+    return user
+
+def create_access_token(data: dict, expires_delta: Union[timedelta, None] = None):
```

Comment:
What data is required to create the access token? Could you add a DocString containing this information please.